frontend/socialapp/my-pov-app/app/screens/OneTimePasswordScreen.tsx

```diff
@@ -0,0 +1,99 @@
+import React, { FC, useState, useRef } from "react"
+import { observer } from "mobx-react-lite"
+import { TextInput, View, Image, TextStyle } from 'react-native';
+import { AppStackScreenProps } from "app/navigators"
+import { Screen, Text } from "app/components"
+
+import { AppStyles } from "app/theme/AppStyles";
+
+const logo = require("../../assets/images/LogoTemp.png");
+
+interface OneTimePasswordScreenProps extends AppStackScreenProps<"OneTimePassword"> {}
+
+export const OneTimePasswordScreen: FC<OneTimePasswordScreenProps> = observer(function OneTimePasswordScreen({
+  route, navigation,
+}) {
+
+  const OTP = "1234";//HARDCODE
+  const [otp, setOtp] = useState(['', '', '', '']);
+  const inputRefs: Array<React.RefObject<TextInput> | null> = [
+    useRef<TextInput>(null),
+    useRef<TextInput>(null),
+    useRef<TextInput>(null),
+    useRef<TextInput>(null),
+  ];
+  
+  const handleChange = (index: number, value: string) => {
+    const newOtp = [...otp];
+
+    if (!isNaN(Number(value)) || value === '') {
```

Comment:
What is this/why do we check like this instead of undefined. i do not know !isNaN. Also, you check if the value is not NaN OR value is EMPTY? doe snot make sense, explain plz